qiskit/aqua/operators/combo_operators/list_op.py

```diff
@@ -0,0 +1,339 @@
+import numpy as np
+from scipy.sparse import spmatrix
+
+from qiskit.circuit import ParameterExpression
+
+from ..operator_base import OperatorBase
+
+
+class ListOp(OperatorBase):
+    """ A class for storing and manipulating lists of operators.
+    Vec here refers to the fact that this class serves
+    as a base class for other Operator combinations which store
+    a list of operators, such as SummedOp or TensoredOp,
+    but also refers to the "vec" mathematical operation.
+    """
+
+    def __init__(self,
+                 oplist: List[OperatorBase],
+                 combo_fn: Callable = lambda x: x,
+                 coeff: Union[int, float, complex, ParameterExpression] = 1.0,
+                 abelian: bool = False) -> None:
+        """
+        Args:
+            oplist: The operators being summed.
+            combo_fn (callable): The recombination function to reduce classical operators
+            when available (e.g. sum)
+            coeff: A coefficient multiplying the operator
+            abelian: indicates if abelian
+
+            Note that the default "recombination function" lambda above is the identity -
+            it takes a list of operators,
+            and is supposed to return a list of operators.
+        """
+        self._oplist = oplist
+        self._combo_fn = combo_fn
+        self._coeff = coeff
+        self._abelian = abelian
+
+    @property
+    def oplist(self) -> List[OperatorBase]:
+        """ returns op list """
+        return self._oplist
+
+    @property
+    def combo_fn(self) -> Callable:
+        """ returns combo function """
+        return self._combo_fn
+
+    @property
+    def abelian(self) -> bool:
+        """ returns abelian """
+        return self._abelian
+
+    # TODO: Keep this property for evals or just enact distribution at composition time?
+    @property
+    def distributive(self) -> bool:
+        """ Indicates whether the ListOp or subclass is distributive under composition.
+        ListOp and SummedOp are,
+        meaning that opv @ op = opv[0] @ op + opv[1] @ op +... (plus for SummedOp,
+        vec for ListOp, etc.),
+        while ComposedOp and TensoredOp do not behave this way."""
+        return True
+
+    @property
+    def coeff(self) -> Union[int, float, complex, ParameterExpression]:
+        """ returns coeff """
+        return self._coeff
+
+    def get_primitives(self) -> set:
+        """ Return a set of strings describing the primitives contained in the Operator """
+        return reduce(set.union, [op.get_primitives() for op in self.oplist])
+
+    @property
+    def num_qubits(self) -> int:
+        """ For now, follow the convention that when one composes to a Vec,
+        they are composing to each separate system. """
+        # return sum([op.num_qubits for op in self.oplist])
+        # TODO maybe do some check here that they're the same?
+        return self.oplist[0].num_qubits
+
+    # TODO change to *other to efficiently handle lists?
```

Comment:
Is the list not unrolled somewhere else? I think it would be better to not change the signature to `*other` to not differ from the parent class

Reply:
Agreed, removing todo